pkg/providers/factory/providerfactory.go

```diff
@@ -71,5 +70,5 @@ func (p *ProviderFactory) BuildProvider(clusterConfigFileName string, clusterCon
 		}
 		return docker.NewProvider(datacenterConfig, p.DockerClient, p.DockerKubectlClient, time.Now), nil
 	}
-	return nil, errors.New("valid providers include: " + constants.DockerProviderName + ", " + constants.VSphereProviderName)
+	return nil, fmt.Errorf("valid providers include: %s, %s, %s, %s, %s", constants.DockerProviderName, constants.VSphereProviderName, constants.CloudStackProviderName, constants.SnowProviderName, constants.TinkerbellProviderName)
```

Comment:
I think it might be more valuable to the end user, who cares about their kinds, to say something like "no provider exists for kind %s", clusterConfig.Spec.DatacenterRef.Kind. Otherwise the connection between provider and type is unclear, as we discovered last night.

Reply:
Additional details in https://github.com/aws/eks-anywhere/issues/1569

Reply:
updated